tests/matrix_multiply_tests/matrix_multiply.futil

```diff
@@ -0,0 +1,232 @@
+import "primitives/core.futil";
+import "primitives/memories/comb.futil";
+import "primitives/binary_operators.futil";
+import "primitives/unsynthesizable.futil";
+
+// Modified from pass-in-register.futil
+component multiply_and_add(
+  // a: 32,
+  // b: 32
+) -> () {
+  cells {
+    ref acc = std_reg(32);
+    ref loaded_val_a = std_reg(32);
+    ref loaded_val_b = std_reg(32);
+    add = std_add(32);
+    mult = std_mult_pipe(32);
+  }
+  wires {
+    group do_mul_and_add {
+      mult.left = loaded_val_a.out;
+      mult.right = loaded_val_b.out;
+      mult.go = !mult.done ? 1'd1;
+
+      add.left = acc.out;
+      add.right = mult.out;
+
+      acc.in = add.out;
+      acc.write_en = mult.done;
+      do_mul_and_add[done] = acc.done;
+    }
+  }
+  control {
+    seq { do_mul_and_add; }
+  }
+}
+
+component main() -> () {
+  cells {
+    // 4×4 matrix memories
+    @external arr_1 = comb_mem_d2(32, 4, 4, 4, 4);
```

Comment:
Sorry, I may have mentioned `comb_mem_d*` in one of our previous meetings, but it's actually better to use `seq_mem_d*` instead! More details in the [docs](https://docs.calyxir.org/libraries/core.html#memories) :)